sdk/src/main/java/com/mapbox/maps/ResourcesAttributeParser.kt

```diff
@@ -16,17 +16,11 @@ internal object ResourcesAttributeParser {
    */
   fun parseResourcesOptions(
     context: Context,
-    typedArray: TypedArray,
-    credentialsManager: CredentialsManager
+    typedArray: TypedArray
   ): ResourceOptions {
-    val builder = ResourceOptionsManager.getDefault(context).resourceOptions.toBuilder()
 
-    if (typedArray.hasValue(R.styleable.mapbox_MapView_mapbox_resourcesAccessToken)) {
-      builder.accessToken(
-        typedArray.getString(R.styleable.mapbox_MapView_mapbox_resourcesAccessToken)
-          ?: credentialsManager.getAccessToken(context)
-      )
-    }
+    val token = typedArray.getString(R.styleable.mapbox_MapView_mapbox_resourcesAccessToken)
+    val builder = ResourceOptionsManager.getDefault(context, token).resourceOptions.toBuilder()
```

Comment:
what if the user has already used the `ResourceOptionsManager.getDefault(context)` previously, the ResourceOptionsManager.default is set, and this method won't use the token from the configured xml attribute.

We should add some tests to validate the correct decision tree.

Reply:
Please refer to this [line](https://github.com/mapbox/mapbox-maps-android/pull/365/files#diff-fcbcc1ee0a08a0b88a933c358ca04796a5731db3cbb0696bc926635783e65f72R110) the current token has the highest priority

Reply:
I meant if the `if (default == null) {` is false, we will not get to the `applyDefaultParams`.